interface/main/finder/multi_patients_finder.php

```diff
@@ -0,0 +1,258 @@
+            background-color: white;
+            overflow: auto;
+        }
+
+        #searchResults .remove-patient {
+            cursor: hand;
+            cursor: pointer;
+        }
+        #searchResults td {
+            /*font-size: 0.7em;*/
+            border-bottom: 1px solid #eee;
+        }
+    </style>
+
+</head>
+
+<body class="body_top">
+<div class="container-responsive">
+    <div id="searchCriteria">
+        <form class="form-inline">
+            <div class="select-box">
+                <label><?php echo xlt('Patient name') .':'; ?></label>
+                <select id="by-name" class="input-sm">
+                    <option value=""><?php echo xlt('Type name'); ?></option>
+                </select>
+                <label><?php echo xlt('Patient ID'); ?></label>
+                <select id="by-id" class="input-sm">
+                    <option value=""><?php echo xlt('Type ID'); ?></option>
+                </select>
+            </div>
+            <div class="buttons-box">
+                <div class="inline-box">
+                    <button id="add-to-list"><?php echo xlt('Add to list'); ?></button>
+                </div>
+                <div class="inline-box">
+                    <button id="send-patients" onclick="selPatients()"><?php echo  xlt('OK'); ?></button>
+                </div>
+            </div>
+        </form>
+    </div>
+
+    <table id="results-table" class="table table-condensed">
+        <thead id="searchResultsHeader" class="head">
+        <tr>
+            <th class="srName"><?php echo  xlt('Name'); ?></th>
+            <th class="srPhone"><?php echo xlt('Phone'); ?></th>
+            <th class="srSS"><?php echo xlt('SS'); ?></th>
+            <th class="srDOB"><?php echo xlt('DOB'); ?></th>
+            <th class="srID"><?php echo xlt('ID'); ?></th>
+            <th></th>
+        </tr>
+        </thead>
+        <tbody id="searchResults">
+        <?php
+        if (isset($_GET['patients'])) {
+            foreach ($results as $index => $result) {
+                echo '<tr id="row'.$index.'">' .
+                        '<td>'. $result['lname'] . ', ' . $result['fname'] . '</td>' .
+                        '<td>' . $result['phone_home'] . '</td>' .
+                        '<td>' . $result['ss'] . '</td>' .
+                        '<td>' . $result['DOB'] . '</td>' .
+                        '<td>' . $result['pubpid'] . '</td>' .
+                        '<td><i class="fa fa-remove remove-patient" onclick="removePatient('.$index.')"></i></td>' .
+                    '<tr>';
+            }
+        } ?>
+        </tbody>
+    </table>
+
+</div>
+
+<script>
+
+var currentResult;
+
+<?php if (isset($_GET['patients'])) { ?>
+var patientsList = <?php echo json_encode($results); ?>;
+<?php } else { ?>
+var patientsList = [];
+$('#results-table').hide();
+<?php } ?>
+
+//Initial select2 library for auto completing using ajax
+$('#by-id, #by-name').select2({
```

Comment:
top.restoreSession()

Reply:
Why this function is needed? haven't saw similar code in the another projects..
Is it for frames layout?

Where I need to put it? in the callback of the response of the ajax? here I haven't callback..

Reply:
Place it before the ajax call. This prevents session leaking (when have multiple tabs of openemr in the same browser) and is discussed here: https://www.open-emr.org/wiki/index.php/OpenEMR_System_Architecture#PHP_Sessions_and_Browser_Windows

Reply:
Ok. thank you.

Reply:
missing the top.restoreSession() here

Reply:
I putted it into the ajax json beforeSend (line 170) (I can't change select2 library)